component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/validator/JWTValidator.java

```diff
@@ -164,6 +169,17 @@ public boolean isValidAssertion(SignedJWT signedJWT) throws OAuthClientAuthnExce
                 issuedTime = issuedAtTime.getTime();
             }
 
+            /* For FAPI compliant applications the allowed JWT signing algorithm should be configured at the
+               application creation and only PS256 and ES256 algorithms are allowed. Therefore these will be checked
+               against the signing algorithm used to sign the JWT in the request.
+               https://openid.net/specs/openid-financial-api-part-2-1_0.html#algorithm-considerations */
+            if (OAuth2Util.isFapiConformantApp(consumerKey)) {
+                if (!isValidSignatureAlgorithm(signedJWT, consumerKey)) {
```

Comment:
Since we are giving the capability to configure the allowed sig alg in the SP we need to that validation regardless of the FAPI conformance.

Reply:
Fixed in f62c438a62cda25d10b408fb266521c3cd30aaec and 393f070cb3d7c0feb53e15516f6e562bddb36dee